Geometry_Engine/Compute/DistributeOutlines.cs

```diff
@@ -194,6 +242,12 @@ private static List<List<PolyCurve>> DistributeOpenings(this List<PolyCurve> pan
         [Output("distributedOutlines", "Lists representing each set of outlines. For each list, the first outline is the containing outline and the remainder are the outlines it contains.")]
         private static List<List<ICurve>> DistributeOpenings(this List<ICurve> panels, List<ICurve> openings, double tolerance = Tolerance.Distance)
         {
+            if (panels == null || panels.Where(x => x != null).Count() == 0 || openings == null || openings.Where(x => x != null).Count() == 0)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot distribute a null list of panels/openings.");
+                return new List<List<ICurve>>();
```

Comment:
Same as my previous comment, shall we return a new collection or `null` in case of null inputs? Is it formalised somewhere @FraserGreenroyd? This would be really helpful to turn this exercise into a nobrainer.

Reply:
https://github.com/BHoM/documentation/wiki/Null-Handling